client/src/analysis/components/RoiPicker.tsx

```diff
@@ -1,88 +1,54 @@
 import * as React from "react";
-import { useState } from "react";
 import { useDispatch } from "react-redux";
-import { AnalysisTypes, SumFrameParams } from "../../messages";
-import { cbToRadius, inRectConstraint, keepOnCY } from "../../widgets/constraints";
-import Disk from "../../widgets/Disk";
-import DraggableHandle from "../../widgets/DraggableHandle";
-import { HandleRenderFunction } from "../../widgets/types";
+
+import { AnalysisTypes, FrameParams } from "../../messages";
 import * as analysisActions from "../actions";
 
-const useRoiPicker = ({ scanWidth, scanHeight, analysisId, enabled, jobIndex }: {
+const useRoiPicker = ({ analysisId, enabled, jobIndex, roiParameters, analys}: {
     scanWidth: number;
     scanHeight: number;
     enabled: boolean;
     jobIndex: number,
     analysisId: string;
+    roiParameters: FrameParams;
+    analys: AnalysisTypes.SD_FRAMES|AnalysisTypes.SUM_FRAMES
 }) => {
-    const minLength = Math.min(scanWidth, scanHeight);
-    const [cx, setCx] = useState(scanWidth / 2);
-    const [cy, setCy] = useState(scanHeight / 2);
-    const [r, setR] = useState(minLength / 8);
 
-    const dispatch = useDispatch();
 
-    const roiParameters: SumFrameParams = {
-        roi: {
-            shape: "disk",
-            cx,
-            cy,
-            r,
-        },
-    }
+    
+    const dispatch = useDispatch();
 
     React.useEffect(() => {
         const handle = setTimeout(() => {
-            if (enabled) {
+            if ((enabled)&&(analys===AnalysisTypes.SD_FRAMES)) {
                 dispatch(analysisActions.Actions.run(analysisId, jobIndex, {
-                    type: AnalysisTypes.SUM_FRAMES,
+                    type: AnalysisTypes.SD_FRAMES,
                     parameters: roiParameters,
                 }))
             }
         }, 100);
 
         return () => clearTimeout(handle);
-    }, [analysisId, enabled, jobIndex, cx, cy, r]);
+    }, [analysisId, enabled, jobIndex, JSON.stringify(roiParameters)]);
```

Comment:
As you are using `analys` inside of the effect function, you also need to add it to the list of dependencies here. Same for `useEffect` below.